comp/core/autodiscovery/integration/config_test.go

```diff
@@ -281,7 +281,9 @@ func TestSetNameForInstance(t *testing.T) {
 
 	err := config.Instances[0].SetNameForInstance("new-name")
 	assert.NoError(t, err)
-	assert.Equal(t, config.Instances[0].GetNameForInstance(), "new-name")
+	expectedInstance := `name: new-name
```

Comment:
Why is this change needed?

Reply:
Understood, got a quick nit otherwise looks good: https://github.com/DataDog/datadog-agent/pull/34513/files#r1973615992